### PR TITLE
Server request factory tests

### PR DESCRIPTION
**Pull request by romkatsu**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | #105

### CODE REVIEW

File: src/ServerRequestFactory.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ private function getUri(array $server): UriInterface
     {
         $uri = $this->uriFactory->createUri();
 
-        if (isset($server['HTTPS']) && !empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off') {
+        if (isset($server['HTTPS']) && !empty($server['HTTPS']) && $server['HTTPS'] !== 'off') {
```
Comment (roxblnfk) [inline src/ServerRequestFactory.php]:
```suggestion
        if (array_key_exists('HTTPS', $server) && $server['HTTPS'] !== '' && $server['HTTPS'] !== 'off') {
```

File: src/ServerRequestFactory.php [hunk 2]
```diff
@@ -68,8 +68,8 @@ public function createFromParameters(array $server, array $headers = [], array $
         }
 
         $protocol = '1.1';
-        if (!empty($_SERVER['SERVER_PROTOCOL'])) {
-            $protocol = str_replace('HTTP/', '', $_SERVER['SERVER_PROTOCOL']);
+        if (!empty($server['SERVER_PROTOCOL'])) {
```
Comment (roxblnfk) [inline src/ServerRequestFactory.php]:
```suggestion
        if (array_key_exists('SERVER_PROTOCOL', $server) && $server['SERVER_PROTOCOL'] !== '') {
```

**Review by roxblnfk — APPROVED** [review]

File: src/ServerRequestFactory.php (hunk 1 above)
Comment (romkatsu) [reply src/ServerRequestFactory.php]: fixed

File: src/ServerRequestFactory.php (hunk 2 above)
Comment (romkatsu) [reply src/ServerRequestFactory.php]: fixed

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**